packages/melos/lib/src/commands/bootstrap.dart

```diff
@@ -199,6 +214,127 @@ mixin _BootstrapMixin on _CleanMixin {
     }
   }
 
+  Future<void> _setSharedPreferencesInPackage(
+    Package package, {
+    required Environment? environment,
+    required Map<String, DependencyReference>? dependencies,
+    required Map<String, DependencyReference>? devDependencies,
+  }) async {
+    final packagePubspecFile = utils.pubspecPathForDirectory(package.path);
+    final packagePubspecContents = await readTextFileAsync(packagePubspecFile);
+    final pubspecEditor = YamlEditor(packagePubspecContents);
+
+    final updatedEnvironment = _updateEnvironment(
+      pubspecEditor: pubspecEditor,
+      workspaceEnvironment: environment,
+      packageEnvironment: package.pubSpec.environment,
+    );
+
+    final updatedDependenciesCount = _updateDependencies(
+      pubspecEditor: pubspecEditor,
+      workspaceDependencies: dependencies,
+      packageDependencies: package.pubSpec.dependencies,
+      pubspecKey: 'dependencies',
+    );
+
+    final updatedDevDependenciesCount = _updateDependencies(
+      pubspecEditor: pubspecEditor,
+      workspaceDependencies: devDependencies,
+      packageDependencies: package.pubSpec.devDependencies,
+      pubspecKey: 'dev_dependencies',
+    );
+
+    await writeTextFileAsync(
+      packagePubspecFile,
+      pubspecEditor.toString(),
+    );
+
+    final message = <String>[
+      if (updatedEnvironment) 'Updated environment',
+      if (updatedDependenciesCount > 0)
+        'Updated $updatedDependenciesCount dependencies',
+      if (updatedDevDependenciesCount > 0)
+        'Updated $updatedDevDependenciesCount dev_dependencies',
+    ];
+    if (message.isNotEmpty) {
+      logger.child(packageNameStyle(package.name), prefix: '').child(
+            message.join('\n'),
```

Comment:
Removing the trailing comma in this line makes the whole log statement format a little nicer.